inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -421,17 +422,19 @@ def refextract(obj, eng):
     with get_document_in_workflow(obj) as tmp_document:
         if tmp_document:
             pdf_references = extract_references_from_pdf(tmp_document, source)
+            pdf_references = match_references(pdf_references)
 
     text = get_value(obj.extra_data, 'formdata.references')
     if text:
         text_references = extract_references_from_text(text, source)
+        text_references = match_references(text_references)
 
     if len(pdf_references) == len(text_references) == 0:
         obj.log.info('No references extracted.')
-    elif len(pdf_references) >= len(text_references):
+    elif len(pdf_references) > len(text_references):
         obj.log.info('Extracted %d references from PDF.', len(pdf_references))
         obj.data['references'] = pdf_references
-    elif len(text_references) > len(pdf_references):
+    elif len(text_references) >= len(pdf_references):
```

Comment:
Can you remind me why this changed behavior (preferring text references to PDF references when they are in the same amount) ?

Reply:
The assumption here is that the text references provided by the submitter potentially have a better quality than references extracted from PDF. So, if both of them manage to return the same amount of references, we prefer the text ones.

Reply:
Ok, makes sense. Nit: let's put this behavior change in a distinct commit, so if @michamos says that we did a mistake we can easily revert it (did we ever get his opinion on this?).

Reply:
We didn't get his opinion on this. But that makes sense.

Reply:
Seems OK to me.